services/QuillLMS/engines/comprehension/test/controllers/comprehension/rules_controller_test.rb

```diff
@@ -179,6 +180,16 @@ class RulesControllerTest < ActionController::TestCase
         assert_equal plagiarism_text, parsed_response['plagiarism_text']['text']
       end
 
+      should "return an error if plagiarism rule already exists for prompt" do
```

Comment:
Is it maybe worth adding a couple more tests to the model itself?
- Make sure that new items are `valid?` if they are the first Plagiarism rule for a prompt
- Make sure that new items are not `valid?` if they are not the first Plagiarism rule for a prompt
- Maybe make sure that creating back to back rules for two different prompts works?

Reply:
good call, I'll add these tests